diff --git a/libraries/stdlib/common/src/generated/_Arrays.kt b/libraries/stdlib/common/src/generated/_Arrays.kt
--- a/libraries/stdlib/common/src/generated/_Arrays.kt
+++ b/libraries/stdlib/common/src/generated/_Arrays.kt
@@ -15257,7 +15257,7 @@ public inline fun CharArray.forEachIndexed(action: (index: Int, Char) -> Unit):
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.7")
-@kotlin.jvm.JvmName("maxOrThrow")
+@kotlin.jvm.JvmName("maxOrThrowBoxed")
 @Suppress("CONFLICTING_OVERLOADS")
 public fun Array<out Double>.max(): Double {
     if (isEmpty()) throw NoSuchElementException()
@@ -15279,7 +15279,7 @@ public fun Array<out Double>.max(): Double {
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.7")
-@kotlin.jvm.JvmName("maxOrThrow")
+@kotlin.jvm.JvmName("maxOrThrowBoxed")
 @Suppress("CONFLICTING_OVERLOADS")
 public fun Array<out Float>.max(): Float {
     if (isEmpty()) throw NoSuchElementException()
@@ -17769,6 +17769,7 @@ public inline fun <R> CharArray.maxOfWithOrNull(comparator: Comparator<in R>, se
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.4")
+@kotlin.clr.ClrName("maxOrNullBoxed")
 public fun Array<out Double>.maxOrNull(): Double? {
     if (isEmpty()) return null
     var max = this[0]
@@ -17787,6 +17788,7 @@ public fun Array<out Double>.maxOrNull(): Double? {
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.4")
+@kotlin.clr.ClrName("maxOrNullBoxed")
 public fun Array<out Float>.maxOrNull(): Float? {
     if (isEmpty()) return null
     var max = this[0]
@@ -18229,7 +18231,7 @@ public fun CharArray.maxWithOrNull(comparator: Comparator<in Char>): Char? {
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.7")
-@kotlin.jvm.JvmName("minOrThrow")
+@kotlin.jvm.JvmName("minOrThrowBoxed")
 @Suppress("CONFLICTING_OVERLOADS")
 public fun Array<out Double>.min(): Double {
     if (isEmpty()) throw NoSuchElementException()
@@ -18251,7 +18253,7 @@ public fun Array<out Double>.min(): Double {
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.7")
-@kotlin.jvm.JvmName("minOrThrow")
+@kotlin.jvm.JvmName("minOrThrowBoxed")
 @Suppress("CONFLICTING_OVERLOADS")
 public fun Array<out Float>.min(): Float {
     if (isEmpty()) throw NoSuchElementException()
@@ -20741,6 +20743,7 @@ public inline fun <R> CharArray.minOfWithOrNull(comparator: Comparator<in R>, se
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.4")
+@kotlin.clr.ClrName("minOrNullBoxed")
 public fun Array<out Double>.minOrNull(): Double? {
     if (isEmpty()) return null
     var min = this[0]
@@ -20759,6 +20762,7 @@ public fun Array<out Double>.minOrNull(): Double? {
  * @sample samples.collections.Collections.Aggregates.maxMinFloating
  */
 @SinceKotlin("1.4")
+@kotlin.clr.ClrName("minOrNullBoxed")
 public fun Array<out Float>.minOrNull(): Float? {
     if (isEmpty()) return null
     var min = this[0]
diff --git a/libraries/stdlib/common/src/generated/_Collections.kt b/libraries/stdlib/common/src/generated/_Collections.kt
--- a/libraries/stdlib/common/src/generated/_Collections.kt
+++ b/libraries/stdlib/common/src/generated/_Collections.kt
@@ -3471,6 +3471,7 @@ public operator fun <T> Iterable<T>.minus(elements: Iterable<T>): List<T> {
 /**
  * Returns a list containing all elements of the original collection except the elements contained in the given [elements] sequence.
  */
+@kotlin.clr.ClrName("minusSequence")
 public operator fun <T> Iterable<T>.minus(elements: Sequence<T>): List<T> {
     val other = elements.toList()
     if (other.isEmpty())
@@ -3578,6 +3579,7 @@ public operator fun <T> Collection<T>.plus(elements: Iterable<T>): List<T> {
 /**
  * Returns a list containing all elements of the original collection and then all elements of the given [elements] sequence.
  */
+@kotlin.clr.ClrName("plusSequence")
 public operator fun <T> Iterable<T>.plus(elements: Sequence<T>): List<T> {
     val result = ArrayList<T>()
     result.addAll(this)
@@ -3588,6 +3590,7 @@ public operator fun <T> Iterable<T>.plus(elements: Sequence<T>): List<T> {
 /**
  * Returns a list containing all elements of the original collection and then all elements of the given [elements] sequence.
  */
+@kotlin.clr.ClrName("plusSequence")
 public operator fun <T> Collection<T>.plus(elements: Sequence<T>): List<T> {
     val result = ArrayList<T>(this.size + 10)
     result.addAll(this)
diff --git a/libraries/stdlib/common/src/generated/_Sequences.kt b/libraries/stdlib/common/src/generated/_Sequences.kt
--- a/libraries/stdlib/common/src/generated/_Sequences.kt
+++ b/libraries/stdlib/common/src/generated/_Sequences.kt
@@ -2963,5 +2963,6 @@ public operator fun <T> Sequence<T>.minus(elements: Iterable<T>): Sequence<T> {
  * The operation is _intermediate_ for this sequence and _terminal_ and _stateful_ for the [elements] sequence.
  */
+@kotlin.clr.ClrName("minusSequence")
 public operator fun <T> Sequence<T>.minus(elements: Sequence<T>): Sequence<T> {
     return object: Sequence<T> {
         override fun iterator(): Iterator<T> {
@@ -3047,6 +3048,7 @@ public operator fun <T> Sequence<T>.plus(elements: Iterable<T>): Sequence<T> {
  *
  * The operation is _intermediate_ and _stateless_.
  */
+@kotlin.clr.ClrName("plusSequence")
 public operator fun <T> Sequence<T>.plus(elements: Sequence<T>): Sequence<T> {
     return sequenceOf(this, elements).flatten()
 }
diff --git a/libraries/stdlib/common/src/generated/_Sets.kt b/libraries/stdlib/common/src/generated/_Sets.kt
--- a/libraries/stdlib/common/src/generated/_Sets.kt
+++ b/libraries/stdlib/common/src/generated/_Sets.kt
@@ -63,5 +63,6 @@ public operator fun <T> Set<T>.minus(elements: Iterable<T>): Set<T> {
  * The returned set preserves the element iteration order of the original set.
  */
+@kotlin.clr.ClrName("minusSequence")
 public operator fun <T> Set<T>.minus(elements: Sequence<T>): Set<T> {
     val result = LinkedHashSet<T>(this)
     result.removeAll(elements)
@@ -122,5 +123,6 @@ public operator fun <T> Set<T>.plus(elements: Iterable<T>): Set<T> {
  * The returned set preserves the element iteration order of the original set.
  */
+@kotlin.clr.ClrName("plusSequence")
 public operator fun <T> Set<T>.plus(elements: Sequence<T>): Set<T> {
     val result = LinkedHashSet<T>(mapCapacity(this.size * 2))
     result.addAll(this)
diff --git a/libraries/stdlib/src/kotlin/collections/Maps.kt b/libraries/stdlib/src/kotlin/collections/Maps.kt
--- a/libraries/stdlib/src/kotlin/collections/Maps.kt
+++ b/libraries/stdlib/src/kotlin/collections/Maps.kt
@@ -600,6 +600,7 @@ public fun <K, V> MutableMap<in K, in V>.putAll(pairs: Iterable<Pair<K, V>>): Un
 /**
  * Puts all the elements of the given sequence into this [MutableMap] with the first component in the pair being the key and the second the value.
  */
+@kotlin.clr.ClrName("putAllSequence")
 public fun <K, V> MutableMap<in K, in V>.putAll(pairs: Sequence<Pair<K, V>>): Unit {
     for ((key, value) in pairs) {
         put(key, value)
@@ -767,12 +768,14 @@ public fun <K, V, M : MutableMap<in K, in V>> Array<out Pair<K, V>>.toMap(destin
  * The returned map preserves the entry iteration order of the original sequence.
  * If any of two pairs would have the same key the last one gets added to the map.
  */
+@kotlin.clr.ClrName("toMapSequence")
 public fun <K, V> Sequence<Pair<K, V>>.toMap(): Map<K, V> = toMap(LinkedHashMap<K, V>()).optimizeReadOnlyMap()

 /**
  * Populates and returns the [destination] mutable map with key-value pairs from the given sequence of pairs.
  */
 @IgnorableReturnValue
+@kotlin.clr.ClrName("toMapSequence")
 public fun <K, V, M : MutableMap<in K, in V>> Sequence<Pair<K, V>>.toMap(destination: M): M =
     destination.apply { putAll(this@toMap) }

@@ -837,6 +840,7 @@ public operator fun <K, V> Map<out K, V>.plus(pairs: Array<out Pair<K, V>>): Map
  * The returned map preserves the entry iteration order of the original map.
  * Those [pairs] with unique keys are iterated in the end in the order of [pairs] sequence.
  */
+@kotlin.clr.ClrName("plusSequence")
 public operator fun <K, V> Map<out K, V>.plus(pairs: Sequence<Pair<K, V>>): Map<K, V> =
     LinkedHashMap(this).apply { putAll(pairs) }.optimizeReadOnlyMap()

@@ -878,6 +882,7 @@ public inline operator fun <K, V> MutableMap<in K, in V>.plusAssign(pairs: Array
  * Appends or replaces all pairs from the given sequence of [pairs] in this mutable map.
  */
 @kotlin.internal.InlineOnly
+@kotlin.clr.ClrName("plusAssignSequence")
 public inline operator fun <K, V> MutableMap<in K, in V>.plusAssign(pairs: Sequence<Pair<K, V>>) {
     putAll(pairs)
 }
@@ -926,6 +931,7 @@ public operator fun <K, V> Map<out K, V>.minus(keys: Array<out K>): Map<K, V> =
  * The returned map preserves the entry iteration order of the original map.
  */
 @SinceKotlin("1.1")
+@kotlin.clr.ClrName("minusSequence")
 public operator fun <K, V> Map<out K, V>.minus(keys: Sequence<K>): Map<K, V> =
     this.toMutableMap().apply { minusAssign(keys) }.optimizeReadOnlyMap()

@@ -961,6 +967,7 @@ public inline operator fun <K, V> MutableMap<K, V>.minusAssign(keys: Array<out K
  */
 @SinceKotlin("1.1")
 @kotlin.internal.InlineOnly
+@kotlin.clr.ClrName("minusAssignSequence")
 public inline operator fun <K, V> MutableMap<K, V>.minusAssign(keys: Sequence<K>) {
     this.keys.removeAll(keys)
 }
diff --git a/libraries/stdlib/src/kotlin/collections/MutableCollections.kt b/libraries/stdlib/src/kotlin/collections/MutableCollections.kt
--- a/libraries/stdlib/src/kotlin/collections/MutableCollections.kt
+++ b/libraries/stdlib/src/kotlin/collections/MutableCollections.kt
@@ -73,6 +73,7 @@ public inline operator fun <T> MutableCollection<in T>.plusAssign(elements: Arra
  * Adds all elements of the given [elements] sequence to this mutable collection.
  */
 @kotlin.internal.InlineOnly
+@kotlin.clr.ClrName("plusAssignSequence")
 public inline operator fun <T> MutableCollection<in T>.plusAssign(elements: Sequence<T>) {
     this.addAll(elements)
 }
@@ -105,6 +106,7 @@ public inline operator fun <T> MutableCollection<in T>.minusAssign(elements: Arr
  * Removes all elements contained in the given [elements] sequence from this mutable collection.
  */
 @kotlin.internal.InlineOnly
+@kotlin.clr.ClrName("minusAssignSequence")
 public inline operator fun <T> MutableCollection<in T>.minusAssign(elements: Sequence<T>) {
     this.removeAll(elements)
 }
@@ -129,6 +131,7 @@ public fun <T> MutableCollection<in T>.addAll(elements: Iterable<T>): Boolean {
  * Adds all elements of the given [elements] sequence to this [MutableCollection].
  */
 @IgnorableReturnValue
+@kotlin.clr.ClrName("addAllSequence")
 public fun <T> MutableCollection<in T>.addAll(elements: Sequence<T>): Boolean {
     var result: Boolean = false
     for (item in elements) {
@@ -164,6 +167,7 @@ public fun <T> MutableCollection<in T>.removeAll(elements: Iterable<T>): Boolean
  * Removes all elements from this [MutableCollection] that are also contained in the given [elements] sequence.
  */
 @IgnorableReturnValue
+@kotlin.clr.ClrName("removeAllSequence")
 public fun <T> MutableCollection<in T>.removeAll(elements: Sequence<T>): Boolean {
     val list = elements.toList()
     return list.isNotEmpty() && removeAll(list)
@@ -200,6 +204,7 @@ public fun <T> MutableCollection<in T>.retainAll(elements: Array<out T>): Boolea
  * Retains only elements of this [MutableCollection] that are contained in the given [elements] sequence.
  */
 @IgnorableReturnValue
+@kotlin.clr.ClrName("retainAllSequence")
 public fun <T> MutableCollection<in T>.retainAll(elements: Sequence<T>): Boolean {
     val list = elements.toList()
     if (list.isNotEmpty())
diff --git a/libraries/stdlib/src/kotlin/collections/SequenceBuilder.kt b/libraries/stdlib/src/kotlin/collections/SequenceBuilder.kt
--- a/libraries/stdlib/src/kotlin/collections/SequenceBuilder.kt
+++ b/libraries/stdlib/src/kotlin/collections/SequenceBuilder.kt
@@ -125,6 +125,7 @@ public abstract class SequenceScope<in T> internal constructor() {
      *
      * @sample samples.collections.Sequences.Building.buildSequenceYieldAll
      */
+    @kotlin.clr.ClrName("yieldAllSequence")
     public suspend fun yieldAll(sequence: Sequence<T>): Unit = yieldAll(sequence.iterator())
 }

diff --git a/libraries/stdlib/src/kotlin/ranges/Ranges.kt b/libraries/stdlib/src/kotlin/ranges/Ranges.kt
--- a/libraries/stdlib/src/kotlin/ranges/Ranges.kt
+++ b/libraries/stdlib/src/kotlin/ranges/Ranges.kt
@@ -272,6 +272,7 @@ public inline operator fun <T, R> R.contains(element: T?): Boolean where T : Com
 )
 @SinceKotlin("1.3")
 @kotlin.internal.InlineOnly
+@kotlin.clr.ClrName("containsClosedAny")
 public inline operator fun <T, R> R.contains(element: T?): Boolean where T : Any, R : ClosedRange<T>, R : Iterable<T> =
     element != null && contains(element)

@@ -283,6 +284,7 @@ public inline operator fun <T, R> R.contains(element: T?): Boolean where T : Any
 @SinceKotlin("2.3")
 @WasExperimental(ExperimentalStdlibApi::class)
 @kotlin.internal.InlineOnly
+@kotlin.clr.ClrName("containsOpenComparable")
 public inline operator fun <T, R> R.contains(element: T?): Boolean where T : Comparable<T>, R : OpenEndRange<T>, R : Iterable<T> =
     element != null && contains(element)

@@ -299,6 +301,7 @@ public inline operator fun <T, R> R.contains(element: T?): Boolean where T : Com
 )
 @SinceKotlin("1.9")
 @kotlin.internal.InlineOnly
+@kotlin.clr.ClrName("containsOpenAny")
 public inline operator fun <T, R> R.contains(element: T?): Boolean where T : Any, R : OpenEndRange<T>, R : Iterable<T> =
     element != null && contains(element)
